nifi-nar-bundles/nifi-salesforce-bundle/nifi-salesforce-processors/src/main/java/org/apache/nifi/processors/salesforce/QuerySalesforceObject.java

```diff
@@ -0,0 +1,486 @@
+            .description("Controller service to handle Oauth2 authorization.")
+            .identifiesControllerService(OAuth2AccessTokenProvider.class)
+            .required(true)
+            .build();
+
+    static final PropertyDescriptor RECORD_WRITER = new PropertyDescriptor.Builder()
+            .name("record-writer")
+            .displayName("Record Writer")
+            .description("Specifies the Controller Service to use for writing out the records")
+            .identifiesControllerService(RecordSetWriterFactory.class)
+            .required(true)
+            .build();
+
+    static final PropertyDescriptor INCLUDE_ZERO_RECORD_FLOWFILES = new PropertyDescriptor.Builder()
+            .name("include-zero-record-flowfiles")
+            .displayName("Include Zero Record FlowFiles")
+            .description("When converting an incoming FlowFile, if the conversion results in no data, "
+                    + "this property specifies whether or not a FlowFile will be sent to the corresponding relationship")
+            .expressionLanguageSupported(ExpressionLanguageScope.NONE)
+            .allowableValues("true", "false")
+            .defaultValue("false")
+            .required(true)
+            .build();
+
+    static final PropertyDescriptor RESPONSE_TIMEOUT = new PropertyDescriptor.Builder()
+            .name("salesforce-http-response-timeout")
+            .displayName("Response Timeout")
+            .description("Max wait time for a response from the Salesforce REST API.")
+            .required(true)
+            .defaultValue("15 secs")
+            .addValidator(StandardValidators.TIME_PERIOD_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .build();
+
+    static final Relationship REL_SUCCESS = new Relationship.Builder()
+            .name("success")
+            .description("For FlowFiles created as a result of a successful query.")
+            .build();
+
+    private static final String LAST_AGE_FILTER = "last_age_filter";
+    private static final String DATE_FORMAT = "yyyy-MM-dd";
+    private static final String TIME_FORMAT = "HH:mm:ss.SSSX";
+    private static final String DATE_TIME_FORMAT = "yyyy-MM-dd'T'HH:mm:ss.SSSZZZZ";
+
+    private volatile SalesforceToRecordSchemaConverter salesForceToRecordSchemaConverter;
+    private volatile SalesforceRestService salesforceRestService;
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        String dateFormat = context.getProperty(DateTimeUtils.DATE_FORMAT).getValue();
+        String timeFormat = context.getProperty(DateTimeUtils.TIME_FORMAT).getValue();
+        String timestampFormat = context.getProperty(DateTimeUtils.TIMESTAMP_FORMAT).getValue();
+
+        salesForceToRecordSchemaConverter = new SalesforceToRecordSchemaConverter(
+                dateFormat,
+                timestampFormat,
+                timeFormat
+        );
```

Comment:
The `context.getProperty()` calls need to be changed since the processor no longer defines properties for those values. Should those date and time format properties also be hard-coded? The earlier version of the processor had configured properties for those formats, but the usage was too broad, as it also applied to the select query. If the incoming JSON has a standard format for dates and times, then the correct values could be specified in the `SalesforceToRecordSchemaConverter` itself.

Reply:
Yes, they should be, indeed. I'll change it. Thanks!